File: android/app/src/main/java/com/fmaldonado/siase/data/network/Exceptions.kt
package com.fmaldonado.siase.data.network

import okhttp3.ResponseBody
import java.io.IOException

open class AppError(val error: String?) : IOException()

class BadInput(error: String?) : AppError(error)

class Unauthorized(error: String?) : AppError(error)

class EmptySearch(error: String?) : AppError(error)

class NotFoundError(error: String?) : AppError(error)

File: ios/siase/data/network/Exceptions.swift
//
//  Exceptions.swift
//  siase
//
//  Created by Fernando Maldonado on 01/03/22.
//

import Foundation

class AppError : Error{
    let message:String
    
    init(message:String){
        self.message = message
    }
    
}

class BadInput : AppError{
    
}

class Unauthorized : AppError{
    
}

class EmptySearch : AppError{
    
}

class NotFoundError : AppError{
    
}
